// costuminputg/dist/costuminputg.es.js
import v, { useState as ie } from "react";
var w = { exports: {} }, g = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var W;
function ue() {
  if (W) return g;
  W = 1;
  var r = Symbol.for("react.transitional.element"), t = Symbol.for("react.fragment");
  function n(o, c, l) {
    var d = null;
    if (l !== void 0 && (d = "" + l), c.key !== void 0 && (d = "" + c.key), "key" in c) {
      l = {};
      for (var f in c)
        f !== "key" && (l[f] = c[f]);
    } else l = c;
    return c = l.ref, {
      $$typeof: r,
      type: o,
      key: d,
      ref: c !== void 0 ? c : null,
      props: l
    };
  }
  return g.Fragment = t, g.jsx = n, g.jsxs = n, g;
}
var R = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var $;
function fe() {
  return $ || ($ = 1, process.env.NODE_ENV !== "production" && function() {
    function r(e) {
      if (e == null) return null;
      if (typeof e == "function")
        return e.$$typeof === ce ? null : e.displayName || e.name || null;
      if (typeof e == "string") return e;
      switch (e) {
        case x:
          return "Fragment";
        case Q:
          return "Profiler";
        case Z:
          return "StrictMode";
        case te:
          return "Suspense";
        case ne:
          return "SuspenseList";
        case oe:
          return "Activity";
      }
      if (typeof e == "object")
        switch (typeof e.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), e.$$typeof) {
          case H:
            return "Portal";
          case ee:
            return (e.displayName || "Context") + ".Provider";
          case K:
            return (e._context.displayName || "Context") + ".Consumer";
          case re:
            var a = e.render;
            return e = e.displayName, e || (e = a.displayName || a.name || "", e = e !== "" ? "ForwardRef(" + e + ")" : "ForwardRef"), e;
          case ae:
            return a = e.displayName || null, a !== null ? a : r(e.type) || "Memo";
          case z:
            a = e._payload, e = e._init;
            try {
              return r(e(a));
            } catch {
            }
        }
      return null;
    }
    function t(e) {
      return "" + e;
    }
    function n(e) {
      try {
        t(e);
        var a = !1;
      } catch {
        a = !0;
      }
      if (a) {
        a = console;
        var s = a.error, i = typeof Symbol == "function" && Symbol.toStringTag && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return s.call(
          a,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          i
        ), t(e);
      }
    }
    function o(e) {
      if (e === x) return "<>";
      if (typeof e == "object" && e !== null && e.$$typeof === z)
        return "<...>";
      try {
        var a = r(e);
        return a ? "<" + a + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function c() {
      var e = P.A;
      return e === null ? null : e.getOwner();
    }
    function l() {
      return Error("react-stack-top-frame");
    }
    function d(e) {
      if (I.call(e, "key")) {
        var a = Object.getOwnPropertyDescriptor(e, "key").get;
        if (a && a.isReactWarning) return !1;
      }
      return e.key !== void 0;
    }
    function f(e, a) {
      function s() {
        Y || (Y = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          a
        ));
      }
      s.isReactWarning = !0, Object.defineProperty(e, "key", {
        get: s,
        configurable: !0
      });
    }
    function b() {
      var e = r(this.type);
      return D[e] || (D[e] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), e = this.props.ref, e !== void 0 ? e : null;
    }
    function X(e, a, s, i, p, m, T, S) {
      return s = m.ref, e = {
        $$typeof: A,
        type: e,
        key: a,
        props: m,
        _owner: p
      }, (s !== void 0 ? s : null) !== null ? Object.defineProperty(e, "ref", {
        enumerable: !1,
        get: b
      }) : Object.defineProperty(e, "ref", { enumerable: !1, value: null }), e._store = {}, Object.defineProperty(e._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(e, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(e, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: T
      }), Object.defineProperty(e, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: S
      }), Object.freeze && (Object.freeze(e.props), Object.freeze(e)), e;
    }
    function k(e, a, s, i, p, m, T, S) {
      var u = a.children;
      if (u !== void 0)
        if (i)
          if (se(u)) {
            for (i = 0; i < u.length; i++)
              N(u[i]);
            Object.freeze && Object.freeze(u);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else N(u);
      if (I.call(a, "key")) {
        u = r(e);
        var O = Object.keys(a).filter(function(le) {
          return le !== "key";
        });
        i = 0 < O.length ? "{key: someKey, " + O.join(": ..., ") + ": ...}" : "{key: someKey}", L[u + i] || (O = 0 < O.length ? "{" + O.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          i,
          u,
          O,
          u
        ), L[u + i] = !0);
      }
      if (u = null, s !== void 0 && (n(s), u = "" + s), d(a) && (n(a.key), u = "" + a.key), "key" in a) {
        s = {};
        for (var C in a)
          C !== "key" && (s[C] = a[C]);
      } else s = a;
      return u && f(
        s,
        typeof e == "function" ? e.displayName || e.name || "Unknown" : e
      ), X(
        e,
        u,
        m,
        p,
        c(),
        s,
        T,
        S
      );
    }
    function N(e) {
      typeof e == "object" && e !== null && e.$$typeof === A && e._store && (e._store.validated = 1);
    }
    var _ = v, A = Symbol.for("react.transitional.element"), H = Symbol.for("react.portal"), x = Symbol.for("react.fragment"), Z = Symbol.for("react.strict_mode"), Q = Symbol.for("react.profiler"), K = Symbol.for("react.consumer"), ee = Symbol.for("react.context"), re = Symbol.for("react.forward_ref"), te = Symbol.for("react.suspense"), ne = Symbol.for("react.suspense_list"), ae = Symbol.for("react.memo"), z = Symbol.for("react.lazy"), oe = Symbol.for("react.activity"), ce = Symbol.for("react.client.reference"), P = _.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, I = Object.prototype.hasOwnProperty, se = Array.isArray, y = console.createTask ? console.createTask : function() {
      return null;
    };
    _ = {
      "react-stack-bottom-frame": function(e) {
        return e();
      }
    };
    var Y, D = {}, M = _["react-stack-bottom-frame"].bind(
      _,
      l
    )(), F = y(o(l)), L = {};
    R.Fragment = x, R.jsx = function(e, a, s, i, p) {
      var m = 1e4 > P.recentlyCreatedOwnerStacks++;
      return k(
        e,
        a,
        s,
        !1,
        i,
        p,
        m ? Error("react-stack-top-frame") : M,
        m ? y(o(e)) : F
      );
    }, R.jsxs = function(e, a, s, i, p) {
      var m = 1e4 > P.recentlyCreatedOwnerStacks++;
      return k(
        e,
        a,
        s,
        !0,
        i,
        p,
        m ? Error("react-stack-top-frame") : M,
        m ? y(o(e)) : F
      );
    };
  }()), R;
}
var U;
function me() {
  return U || (U = 1, process.env.NODE_ENV === "production" ? w.exports = ue() : w.exports = fe()), w.exports;
}
var E = me(), G = {
  color: void 0,
  size: void 0,
  className: void 0,
  style: void 0,
  attr: void 0
}, J = v.createContext && /* @__PURE__ */ v.createContext(G), de = ["attr", "size", "title"];
function be(r, t) {
  if (r == null) return {};
  var n = pe(r, t), o, c;
  if (Object.getOwnPropertySymbols) {
    var l = Object.getOwnPropertySymbols(r);
    for (c = 0; c < l.length; c++)
      o = l[c], !(t.indexOf(o) >= 0) && Object.prototype.propertyIsEnumerable.call(r, o) && (n[o] = r[o]);
  }
  return n;
}
function pe(r, t) {
  if (r == null) return {};
  var n = {};
  for (var o in r)
    if (Object.prototype.hasOwnProperty.call(r, o)) {
      if (t.indexOf(o) >= 0) continue;
      n[o] = r[o];
    }
  return n;
}
function h() {
  return h = Object.assign ? Object.assign.bind() : function(r) {
    for (var t = 1; t < arguments.length; t++) {
      var n = arguments[t];
      for (var o in n)
        Object.prototype.hasOwnProperty.call(n, o) && (r[o] = n[o]);
    }
    return r;
  }, h.apply(this, arguments);
}
function q(r, t) {
  var n = Object.keys(r);
  if (Object.getOwnPropertySymbols) {
    var o = Object.getOwnPropertySymbols(r);
    t && (o = o.filter(function(c) {
      return Object.getOwnPropertyDescriptor(r, c).enumerable;
    })), n.push.apply(n, o);
  }
  return n;
}
function j(r) {
  for (var t = 1; t < arguments.length; t++) {
    var n = arguments[t] != null ? arguments[t] : {};
    t % 2 ? q(Object(n), !0).forEach(function(o) {
      ve(r, o, n[o]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(r, Object.getOwnPropertyDescriptors(n)) : q(Object(n)).forEach(function(o) {
      Object.defineProperty(r, o, Object.getOwnPropertyDescriptor(n, o));
    });
  }
  return r;
}
function ve(r, t, n) {
  return t = Ee(t), t in r ? Object.defineProperty(r, t, { value: n, enumerable: !0, configurable: !0, writable: !0 }) : r[t] = n, r;
}
function Ee(r) {
  var t = Oe(r, "string");
  return typeof t == "symbol" ? t : t + "";
}
function Oe(r, t) {
  if (typeof r != "object" || !r) return r;
  var n = r[Symbol.toPrimitive];
  if (n !== void 0) {
    var o = n.call(r, t);
    if (typeof o != "object") return o;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (t === "string" ? String : Number)(r);
}
function V(r) {
  return r && r.map((t, n) => /* @__PURE__ */ v.createElement(t.tag, j({
    key: n
  }, t.attr), V(t.child)));
}
function B(r) {
  return (t) => /* @__PURE__ */ v.createElement(ge, h({
    attr: j({}, r.attr)
  }, t), V(r.child));
}
function ge(r) {
  var t = (n) => {
    var {
      attr: o,
      size: c,
      title: l
    } = r, d = be(r, de), f = c || n.size || "1em", b;
    return n.className && (b = n.className), r.className && (b = (b ? b + " " : "") + r.className), /* @__PURE__ */ v.createElement("svg", h({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, n.attr, o, d, {
      className: b,
      style: j(j({
        color: r.color || n.color
      }, n.style), r.style),
      height: f,
      width: f,
      xmlns: "http://www.w3.org/2000/svg"
    }), l && /* @__PURE__ */ v.createElement("title", null, l), r.children);
  };
  return J !== void 0 ? /* @__PURE__ */ v.createElement(J.Consumer, null, (n) => t(n)) : t(G);
}
function Re(r) {
  return B({ attr: { viewBox: "0 0 640 512" }, child: [{ tag: "path", attr: { d: "M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L525.6 386.7c39.6-40.6 66.4-86.1 79.9-118.4c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C465.5 68.8 400.8 32 320 32c-68.2 0-125 26.3-169.3 60.8L38.8 5.1zm151 118.3C226 97.7 269.5 80 320 80c65.2 0 118.8 29.6 159.9 67.7C518.4 183.5 545 226 558.6 256c-12.6 28-36.6 66.8-70.9 100.9l-53.8-42.2c9.1-17.6 14.2-37.5 14.2-58.7c0-70.7-57.3-128-128-128c-32.2 0-61.7 11.9-84.2 31.5l-46.1-36.1zM394.9 284.2l-81.5-63.9c4.2-8.5 6.6-18.2 6.6-28.3c0-5.5-.7-10.9-2-16c.7 0 1.3 0 2 0c44.2 0 80 35.8 80 80c0 9.9-1.8 19.4-5.1 28.2zm9.4 130.3C378.8 425.4 350.7 432 320 432c-65.2 0-118.8-29.6-159.9-67.7C121.6 328.5 95 286 81.4 256c8.3-18.4 21.5-41.5 39.4-64.8L83.1 161.5C60.3 191.2 44 220.8 34.5 243.7c-3.3 7.9-3.3 16.7 0 24.6c14.9 35.7 46.2 87.7 93 131.1C174.5 443.2 239.2 480 320 480c47.8 0 89.9-12.9 126.2-32.5l-41.9-33zM192 256c0 70.7 57.3 128 128 128c13.3 0 26.1-2 38.2-5.8L302 334c-23.5-5.4-43.1-21.2-53.7-42.3l-56.1-44.2c-.2 2.8-.3 5.6-.3 8.5z" }, child: [] }] })(r);
}
function _e(r) {
  return B({ attr: { viewBox: "0 0 576 512" }, child: [{ tag: "path", attr: { d: "M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z" }, child: [] }] })(r);
}
const he = ({ value: r, onChange: t, placeholder: n, label: o, type: c }) => {
  const [l, d] = ie(!1), f = () => {
    d(!l);
  };
  return /* @__PURE__ */ E.jsxs("div", { children: [
    /* @__PURE__ */ E.jsx("label", { className: "text-[13px] text-state-800", children: o }),
    /* @__PURE__ */ E.jsxs("div", { className: "flex w-full justify-between gap-3 text-sm text-black bg-slate-100/50 rounded px-4 py-3 mb-4 mt-3 border border-slate-200 outline-none", children: [
      /* @__PURE__ */ E.jsx(
        "input",
        {
          type: c == "password" ? l ? "text" : "password" : c,
          placeholder: n,
          className: "w-full bg-transparent outline-none",
          value: r,
          onChange: (b) => t(b)
        }
      ),
      c == "password" && /* @__PURE__ */ E.jsx("span", { className: "text-[18px] text-state-800 cursor-pointer", onClick: f, children: l ? /* @__PURE__ */ E.jsx(_e, {}) : /* @__PURE__ */ E.jsx(Re, {}) })
    ] })
  ] });
};
export {
  he as default
};
